src/states/adult/mod.rs

```diff
@@ -552,7 +553,8 @@ impl Base for Adult {
 
     fn handle_timeout(&mut self, token: u64, _: &mut dyn EventBox) -> Transition {
         if self.parsec_timer_token == token {
-            self.send_parsec_poke();
+            // TODO: send this only when the knowledge changes, not periodically.
```

Comment:
We need it to poke regularly though. Only half way through the combined change though so I may be missing something, but is it really sensible to use the same message for Poke and send_member_knowledge?

Reply:
It's not implemented in this PR, but I was thinking it should be sufficient to send this message only when our parsec version knowledge changes. That happens once when we transition from `JoiningPeer` to `Adult` and then every time we receive `GenesisUpdate`. So maybe we should only send it then? I'm not 100% sure about it which is why I haven't implemented it in this PR.

Reply:
I agree, this is what should happen for this new message.
But how does that support what poke was doing. i.e why could we do that for poke, will the gossip flow properly even if it is sent only once, at a point the recipient elder may be in any state. If so that is good.

Reply:
I think it should work. The only problem I can think of is what happens if some of the `MemberKnowledge` messages get lost. Because they would be sent only once, the elders that didn't receive them wouldn't get their members knowledge updated. Not sure how big of a problem it would really be, because we can expect that at least some of the elders do receive the message and those elders would then send the correct version gossip to the node.

Reply:
It is a direct message so it would not get lost (i.e we will retry to send it until it succeed).
However, in the case of a node becoming an elder, they need to keep sending poke until the elders decide that that Adult is about to become an Elder, right in the DKG, so if there is nothing done for that that could be a problem (though maybe it is not an issue, the new node would be a gossip recipient and receive the right version of parsec gossip probably). (Though there may still be a problem if Elders move forward and send new version of gossip where Adult would be an Elder, but the Adult has not caught up yet, in this case poke make sense.

It just seem we are not 100% clear what should be done here.

Reply:
OK. I'm thinking about putting the parsec poke thing back exactly as it was and adding the `MemberKnowledge` as a separate message. Then we can decide what to do with the poke later and address it in a separate PR. What do you think?

Reply:
Not sure, that discussion moved me a bit closer to thinking that these 2 messages should be as 1 as you did. still need to think about it a bit.